software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,640 @@
+#
+#   vanilla_stats_parser.py
+#
+#   vanilla core stats extractor
+# 
+#   input: vanilla_stats.log
+#   output: manycore_stats.log
+#   output: tile_stats/tile_#_stats.log for all tiles 
+#
+#   @author Borna
+#
+#   How to use:
+#   python3 generate_stats.py --dim-y {manycore_dim_y}  --dim-x {manycore_dim_x} 
+#                             --tile (optional) --tile_group (optional)
+#                             --input {vanilla_stats.log}
+#
+#   ex) python3 --input generate_stats.py --dim-y 4 --dim-x 4 --tile --tile_group --input vanilla_stats.log
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore default = 4
+#   {manycore_dim_x}  Mesh X dimension of manycore default = 4
+#   {per_tile}        Generate separate stats file for each tile default = False
+#   {input}           Vanilla stats input file     default = vanilla_stats.log
+
+
+
+import sys
+import argparse
+import os
+import re
+import csv
+from enum import Enum
+from collections import Counter
+
+
+
+
+
+
+# These values are used by the manycore library in bsg_print_stat instructions
+# they are added to the tag value to determine the tile group that triggered the stat
+# and also the type of stat (stand-alone stat, start, or end)
+# the value of these paramters should match their counterpart inside 
+# bsg_manycore/software/bsg_manycore_lib/bsg_manycore.h
+# Formatting for bsg_cuda_print_stat instructions
+# < stat_type >
+# Section                 Stat type  -   y cord   -   x cord   -    tile group id   -        tag
+# of bits                <----2----> -   <--6-->  -   <--6-->  -   <------10----->  -   <-----8----->
+# Stat type value: {"stat":0, "start":1, "end":2}
+bsg_STAT_TAG_BITS   = 8
```

Comment:
A more object-oriented way to do this is to define a CudaTag class. When you read in the CSV, read each tag value and initialize it as a CudaTag object.

Specifically, these variables become class fields (i.e. not initialized in __init__). 

That way, CudaTags can be queried for different properties. For example tag.isStart() can be used to determine if a tag is a start tag. Ditto for finish, or just a normal tag, or the tile group ID.

Reply:
@tommydcjung Do we have a standard naming for bit-width fields? 

@bornaehsani I feel that the better naming here is _width

Reply:
That's a great idea, I created a CudaStatTag class to be used this way. I also changed instances of _BITS to _WIDTH in parameters. Let me know what you think!